EIPS/eip-5169.md

```diff
@@ -0,0 +1,191 @@
+
+#### Overview
+
+With the discussion above in mind, we outline the solution proposed by this ERC. For this purpose, we consider the following variables:
+
+- `SCPrivKey`: The private signing key to administrate a smart contract implementing this ERC. Note that this doesn't have to be a new key especially added for this ERC. Most smart contracts made today already have an administration key to manage the tokens issued. It can be used to update the `scriptURI`.
+
+- `newScriptURI`: an array of URIs for different ways to find the client script.
+
+We can describe the life cycle of the `scriptURI` functionality:
+
+- Issuance
+
+1. The token issuer issues the tokens and a smart contract implementing this ERC, with the admin key for the smart contract being `SCPrivKey`.
+2. The token issuer calls `setScriptURI` with the `scriptURI`.
+
+- Update `scriptURI`
+
+1. The token issuer stores the desired `script` at all the new URI locations and constructs a new `scriptURI` structure based on this. 
+2. The token issuer calls `setScriptURI` with the new `scriptURI` structure.
+
+### Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY” and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+We define a scriptURI element using the `string[]`.
+Based on this, we define the smart contract interface below:
+```
+interface IERC5XX0 {
+    /// @dev This event emits when the scriptURI is updated, 
+    /// so wallets implementing this interface can update a cached script
+    event ScriptUpdate(string[] memory newScriptURI);
+
+    /// @notice Get the scriptURI for the contract
+    /// @return The scriptURI
+    function scriptURI() external view returns(string[] memory);
+
+    /// @notice Update the scriptURI 
+    /// emits event ScriptUpdate(scriptURI memory newScriptURI);
+    function setScriptURI(string[] memory newScriptURI) external;
+}
+```
+The interface MUST be implemented under the following constraints:
+
+- The smart contract implementing `IERC5XX0` MUST store variables `address owner` in its state.
+
+- The smart contract implementing `IERC5XX0` MUST set `owner=msg.sender` in its constructor.
+
+- The `ScriptUpdate(...)` event MUST be emitted when the ```setScriptURI``` function updates the `scriptURI`.
+
+- The `setScriptURI(...)` function MUST validate that `owner == msg.sender` *before* executing its logic and updating any state.
+
+- The `setScriptURI(...)` function MUST update its internal state such that `currentScriptURI = newScriptURI`.
+
+- The `scriptURI()` function MUST return the `currentScriptURI` state.
+
+- The `scriptURI()` function MAY be implemented as pure or view.
+
+- Any user of the script learned from `scriptURI` MUST validate the script is either at an immutable location, its URI contains its hash digest, or it implements ERC 5XX1, which asserts authenticity using signatures instead of a digest.
```

Comment:
```suggestion
- Any user of the script learned from `scriptURI` MUST validate the script is either at an immutable location, its URI contains its hash digest, or it implements `IERC5169`, which asserts authenticity using signatures instead of a digest.
```